Refactor how the generif md5 sum is calculated and stored in XAPIAN.

BonfaceKilz opened the pull request:
This PR re-works how the generif checksums are built in XAPIAN.  Instead of building it straight from virtuoso, we build the md5sum straight from the ttl file.

BonfaceKilz committed:
Refactor how the generif md5 sum is calculated and stored in XAPIAN.

* scripts/index-genenetwork (hash_generif_graph): Build the generif
checksum by directly building it from the file.
(is_data_modified): Update how generif-checksums are verified.
(create_xapian_index): Update how generif-checksums are stored in
XAPIAN.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

jnduli commented on scripts/index-genenetwork:
```diff
@@ -549,7 +525,11 @@ def is_data_modified(xapian_directory: str,
         # Return a zero exit status code when the data has changed;
         # otherwise exit with a 1 exit status code.
-        if (db.get_metadata("generif-checksum").decode() == hash_generif_graph(sparql_uri) and
+        generif = pathlib.Path("/var/lib/data/generif-metadata.ttl")
```
hardcoding this here makes it impossible to test locally or if our system ever changes storage paths. Can we pass this in as a cmd arg?

jnduli commented on scripts/index-genenetwork:
```diff
@@ -239,35 +240,10 @@ def build_rdf_cache(sparql_uri: str, query: str, remove_common_words: bool = Fal
-        raise TypeError(f"Expected results to be a dict but found {type(results)}")
-    bindings = results["results"]["bindings"]
-    return bindings[0]["hash"]["value"]
+def hash_generif_graph(generif_file: str):
```
1. minor: add typehints for the return type
2. minor: change generif_file type to pathlib.Path
2. can we instead pass in a folder and generate the checksum for all ttl files there? This way, if we ever need to add an index for another file, we don't need to touch checksums

That would be something like:

```
ttl_hash = hashlib.new("md5")
for file in generif_file.glob("*.ttl"):
   with open(file, encoding="utf-8") as f:
       data = f.read()
        ttl_hash.update(data)
return ttl_hash.hexdigest()
```

BonfaceKilz replied:
See cf90b87bab9e for the resolution.

jnduli commented on scripts/index-genenetwork:
```diff
@@ -549,7 +525,11 @@ def is_data_modified(xapian_directory: str,
-        if (db.get_metadata("generif-checksum").decode() == hash_generif_graph(sparql_uri) and
+        generif = pathlib.Path("/var/lib/data/generif-metadata.ttl")
+        generif_checksum = "-1"
+        if generif.exists():
```
I think we can move this check to the `hash_generif_graph` and return "-1" if it doesn't exist. This prevents duplicating this logic

BonfaceKilz replied:
See: ff96efc26c5f

BonfaceKilz committed:
Add type-hints to hash_generif_graph.

* scripts/index-genenetwork (hash_generif_graph): Add proper type
hints.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Generate a checksum for all the ttl files.

* scripts/index-genenetwork (hash_generif_graph): Rename to
hash_rdf_graph.  Generate a checksum of all the turtle files inside
the ttl directory that's the basis for the GN virtuoso graph.
(create_xapian_index): Rename hash_generif_graph -> hash_rdf_graph.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Return a "-1" if the turtle directory does not exist.

* scripts/index-genenetwork (hash_rdf_graph): Remove check for the
turtle directory.
(is_data_modified): Ditto.
(create_xapian_index): Ditto.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>
